pkg/policy/l4.go

```diff
@@ -725,8 +775,14 @@ func (l4 L4PolicyMap) containsAllL3L4(labels labels.LabelArray, ports []*models.
 
 	// Check L3-only filters first.
 	filter, match := l4[api.PortProtocolAny]
-	if match && filter.matchesLabels(labels) {
-		return api.Allowed
+	if match {
+		matches, isDeny := filter.matchesLabels(labels)
+		if matches {
+			if isDeny {
+				return api.Denied
+			}
+			return api.Allowed
+		}
```

Comment:
```suggestion
		switch {
		case matches && isDeny:
			return api.Denied
		case matches:
		        return api.Allowed
		}
```

Reply:
@joestringer It does :smile:! https://tour.golang.org/flowcontrol/10

Reply:
You may have been thinking about the `select` block, which randomizes the case selection if all the cases are "ready" in the case of channels. I sometimes confuse them too, they look similar and have the same length :sweat_smile:

Reply:
They're functionally so similar that I usually assume they're the same thing. Thanks for sharing the link, this helps a lot :)